index.js

```diff
@@ -1,88 +1,135 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
 
-// console.log(counter) 
+// const counter = 2
+
+// var counter = 1
+// var counter = 2
+
+let counter = 1
+counter = 2
+
+console.log("%cQuestion 1", "color: red") 
+
+console.log(counter) 
 // // => 2
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
+const name = "Yu"
+
+console.log("%cQuestion 2", "color: red") 
+
+console.log(name)
 // => "Raffy"
 // name = "Not Raffy" 
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
+function drinkWater(currentThirstLevel) {
+  console.log("Man, I sure am thirsty")
+  currentThirstLevel -= 1
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
+
+console.log("%cQuestion 3", "color: red") 
+
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
 
-// console.log(sameSameButDifferent(5, 5)) 
+function sameSameButDifferent(num, maybeNum) {
+  if (num === maybeNum){
+  
+    return "same same"
+  }
+  else{
+
+    return "same same (but different)"       
+}
+}
+console.log("%cQuestion 4", "color: red") 
+
+console.log(sameSameButDifferent(5, 5)) 
```

Comment:
Very close but not quite. We will need an if, else if and else to cover all the conditionals

```
function sameSameButDifferent(num, maybeNum) {
  if (num === maybeNum) {
    console.log("same same")
  } else if (num == maybeNum) {
    console.log("same same but different")
  } else {
    console.log("different")
  }
}
```